src/test/java/org/hobbit/utils/rdf/RdfHelperTest.java

```diff
@@ -116,6 +118,179 @@ public void testGetStringValue() {
+                "1");
+        model.add(model.getResource("http://example.org/example1"), model.getProperty("http://example.org/property2"),
+                "32768");
+        model.add(model.getResource("http://example.org/example1"), model.getProperty("http://example.org/property3"),
+                "2");
+        model.add(model.getResource("http://example.org/example1"), model.getProperty("http://example.org/property3"),
+                "3");
+        model.add(model.getResource("http://example.org/example1"), model.getProperty("http://example.org/property4"),
+            model.getResource("http://example.org/example1"));
+
+        // literal object matches
+        Assert.assertEquals(1, (int) RdfHelper.getIntValue(model, model.getResource("http://example.org/example1"),
+                model.getProperty("http://example.org/property1")));
+        // resource object matches
+        Assert.assertEquals(32768, (int) RdfHelper.getIntValue(model,
+                model.getResource("http://example.org/example1"), model.getProperty("http://example.org/property2")));
+        // more than one triple matches
+        Assert.assertTrue((new HashSet<Integer>(Arrays.asList(2, 3))).contains(RdfHelper.getIntValue(model,
+                model.getResource("http://example.org/example1"), model.getProperty("http://example.org/property3"))));
+
+        // resource wildcard
+        Assert.assertTrue((new HashSet<Integer>(Arrays.asList(2, 3)))
+                .contains(RdfHelper.getIntValue(model, null, model.getProperty("http://example.org/property3"))));
+        // property wildcard
+
+        Assert.assertTrue((new HashSet<Integer>(Arrays.asList(1, 32768, 2, 3)))
+                .contains(RdfHelper.getIntValue(model, model.getResource("http://example.org/example1"), null)));
+
+
+        // resource and property exist but there is no matching triple
+        Assert.assertNull(RdfHelper.getIntValue(model, model.getResource("http://example.org/example2"),
+                model.getProperty("http://example.org/property1")));
+        // resource does not exist
+        Assert.assertNull(RdfHelper.getIntValue(model, model.getResource("http://example.org/example3"),
+                model.getProperty("http://example.org/property1")));
+        // property does not exist
+        Assert.assertNull(RdfHelper.getIntValue(model, model.getResource("http://example.org/example1"),
+                model.getProperty("http://example.org/property4")));
+        // model is null
+        Assert.assertNull(RdfHelper.getIntValue(null, model.getResource("http://example.org/example1"),
+                model.getProperty("http://example.org/property1")));
+
+        // object is resource instead of Integer
+        Assert.assertNull(RdfHelper.getIntValue(model, model.getResource("http://example.org/example1"), model.getProperty("http://example.org/property4")));
```

Comment:
Please add a test case in which you have a boolean or double value as object instead of the expected integer value.

Reply:
Yes, we are yet to add that. We are working on it!

Reply:
When we try to use a literal of a different datatype than expected and assert Null on it, the test case passes for sure but it also throws an exception of wrong data type format, for e.g., `NumberFormatException.` etc. So, instead of just asserting `null`, should we try something like this?

`Throwable e = null;`
`try {`
            `RdfHelper.getIntValue(model, model.getResource("http://example.org/example1"),model.getProperty("http://example.org/property5"));` //the literal is a boolean in this case
        `} catch (Throwable ex) {`
           ` e = ex;  }`
`        Assert.assertTrue(e instanceof NumberFormatException);
`
The test case works fine for this as well.

Reply:
It seems like there is a misunderstanding. The method can do only one of both—either it returns `null` or it throws an exception. I would suggest to go with `null` and log the exception (as it is implemented for the other methods).